src/utils/makeReactNativeConfig.js

```diff
@@ -226,14 +224,45 @@ function makeReactNativeConfig(
         : userWebpackConfig
     );
 
-    // For simplicity, we don't require users to extend
-    // default config.entry but do it for them.
-    config.entry = defaultWebpackConfig.entry.concat(config.entry);
+    config.entry = injectPolyfillIntoEntry(config.entry, polyfillPath);
 
     return config;
   });
 
   return [configs, PLATFORMS];
 }
 
-module.exports = makeReactNativeConfig;
+/*
+ * Takes user entries from webpack.haul.js,
+ * change them to multi-point entries
+ * and injects polyfills
+ */
+function injectPolyfillIntoEntry(
+  userEntry: WebpackEntry,
+  polyfillPath: string
+): WebpackEntry {
+  if (typeof userEntry === 'string') {
+    return [polyfillPath, userEntry];
+  }
+  if (Array.isArray(userEntry)) {
+    return [polyfillPath, ...userEntry];
+  }
+  if (typeof userEntry === 'object') {
```

Comment:
I'm seeing this logic in 3rd fn already. How about creating some pattern-matching helper, and have it in one place? E.g.:
```js
return selectEntry(userEntry, {
  string: [polyfillPath, userEntry],
  array: [polyfillPath, ...userEntry],
  object: () => {...}
})
```
This could be done as a followup and labelled as a `good first issue` 🙂

Reply:
Yeah, let's do it later.